parser/src/parser.rs

```diff
@@ -73,6 +73,50 @@ pub fn parse_expression(source: &str) -> Result<ast::Expression, ParseError> {
     do_lalr_parsing!(source, Expression, StartExpression)
 }
 
+pub fn parse(source: &str, mode: Mode) -> Result<ast::Top, ParseError> {
```

Comment:
I suggest to move this function to the `builtins.rs` in the vm crate.

Reply:
Yes, agreed, but in any case, this function should reside in the vm crate, since it uses a compilation mode.